sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -88,6 +88,7 @@ private[csv] class CSVOptions(
 
   val quote = getChar("quote", '\"')
   val escape = getChar("escape", '\\')
+  val escapeEscape = getChar("escapeEscape", quote)
```

Comment:
@ep1804, let's use `'\0'` by default as it is the default for `setCharToEscapeQuoteEscaping` if not set according to https://github.com/uniVocity/univocity-parsers/blob/e34a149077147de7cafcafbbaf8a4310e6297584/src/main/java/com/univocity/parsers/csv/CsvFormat.java#L160-L176 because I am worried of changing the existing behaviour for lower versions. Maybe, it is okay for now if it is configurable.